ctx.go

```diff
@@ -235,6 +235,29 @@ func (c *Ctx) UsePrivateKey(key PrivateKey) error {
 	return nil
 }
 
+// UserCertAndKey configures the context to use the given certificate
+// and private key for the SSL handshakes.
+// It allows you to use private keys that are never accessible directly
+// e.g.: to which openssl has access only via Engine module.
+// https://www.openssl.org/docs/man1.1.1/man3/SSL_CTX_use_cert_and_key.html
+func (c *Ctx) UseCertAndKey(cert *Certificate, key *PrivateKey) error {
```

Comment:
Suggestion: 
If the key is going to always be nil in the one case, is an explicit wrapper function a good idea? (ie, one that always has nil as the second param?

Reply:
sure. done the split.